### ISSUE TITLE
Regent: error in handling of global C variables

### ISSUE THREAD

**Issue by mariodirenzo** [StanfordLegion/legion#608]
Regent fails to compile the following code
```
import "regent"

local T = terralib.includecstring [[
   const char * string = "test";
   const char * getString() { return "test2"; };
]]

task main()
   -- This does not work
   var t1 = regentlib.string(T.string)
   -- This works
   var t2 = regentlib.string([T.getString]())
end

regentlib.start(main)
```
producing the error
```
test3.rg:10: unable to specialize value of type table
   var t1 = regentlib.string(T.string)
```

The code produces the correct result if the function ` getString()` is called to generate the string variable.

**Comment by elliottslaughter** [comment]
Should be fixed in 2ed0b16ac7375e7b4ff35ac548eb46ba484686d0. Please note that global variables will be read-only in Regent since we can't guarantee that concurrent access will be safe.

**Comment by mariodirenzo** [comment]
ok, thank you
